flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/TieredStorageMemoryManagerImpl.java

```diff
@@ -200,6 +201,20 @@ public int numOwnerRequestedBuffer(Object owner) {
         return numRequestedBuffer == null ? 0 : numRequestedBuffer.get();
     }
 
+    @Override
+    public void transferBufferOwnership(Object oldOwner, Object newOwner, Buffer buffer) {
+        checkState(buffer.isBuffer(), "Only buffer supports transfer ownership.");
+        AtomicInteger numOldOwnerRequested = numOwnerRequestedBuffers.get(oldOwner);
+        if (numOldOwnerRequested == null) {
+            throw new RuntimeException("Failed to transfer buffer ownership for " + oldOwner);
+        }
```

Comment:
This can be replaced by `checkNotNull`.

Reply:
Use `decNumRequestedBuffer(oldOwner)` instead.